src/views/RoomDetail.vue

```diff
@@ -49,18 +64,90 @@ export default {
   created () {
     this.roomID = this.$route.params.roomID
   },
-  mounted () {
-    const db = this.$firebase.firestore()
-    db
-      .collection('Rooms')
-      .doc(this.roomID)
-      .get()
-      .then(doc => {
-        this.room = doc.data()
-      })
+  async mounted () {
+    await this.getRoomKeyValuePairs()
+    await this.getRoomCheckIns()
+    await this.getRoomCheckInsTomorrow()
+    await this.fetchUserIDList()
+    await this.protection()
   },
   methods: {
-    totalSteps: function () {
+    getRoomKeyValuePairs: async function () {
+      await $db()
+        .collection('Rooms')
+        .doc(this.roomID)
+        .get()
+        .then(doc => {
+          this.room = doc.data()
+          this.room.checkIns = []
+          this.room.checkInsTomorrow = []
+        })
+    },
+    getRoomCheckIns: async function () {
+      await $db()
+        .collection('Rooms/' + this.roomID + '/CheckIn')
+        .where('date', '==', this.currentDate)
+        .get()
+        .then(querySnapshot => {
+          querySnapshot.forEach((checkIn) => {
+            this.room.checkIns = checkIn.data()
+          })
+        })
+    },
+    getRoomCheckInsTomorrow: async function () {
+      await $db()
+        .collection('Rooms/' + this.roomID + '/CheckIn')
+        .where('date', '==', this.dateTomorrow())
+        .get()
+        .then(querySnapshot => {
+          querySnapshot.forEach((checkIn) => {
+            this.room.checkInsTomorrow = checkIn.data()
+          })
+        })
+    },
+    fetchUserIDList: async function () {
+      if (this.room.checkIns.user) {
+        this.room.checkIns.user.forEach(user => this.userIdList.todayIDList.push(user.id))
+      }
+      if (this.room.checkInsTomorrow.user) {
+        this.room.checkInsTomorrow.user.forEach(user => this.userIdList.tomorrowIDList.push(user.id))
+      }
+    },
+    protection: async function () {
+      await this.signInProtection(this.currentDate, this.room.checkIns.user)
+      await this.signInProtection(this.dateTomorrow(), this.room.checkInsTomorrow.user)
+    },
+    signInProtection: async function (date, room) {
+      let noCapacity = false
+      const isSignedIn = await this.getSignedRoom(date)
+      if (room) {
+        if (room.length >= this.roomCapacity()) {
+          noCapacity = true
+        }
+      } if (isSignedIn || noCapacity) {
+        if (room === this.room.checkIns.user) {
```

Comment:
Hier ist es klüger das Datum zu prüfen, statt die Räumen
```suggestion
        if (date === this.currentDate) {
```

Reply:
Wieso? :) Geht um die Auswahl des richtigen Buttons.